include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -341,19 +344,14 @@ class pool_memory_resource final
    * @param stream The stream on which the memory is to be used.
    * @return block_type The allocated block
```

Comment:
This function might now throw, could you add a note to that effect in the docstring?